seed/php-sdk/imdb/clientName/src/Core/Client/RetryMiddleware.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace Seed\Core\Client;
+
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Promise as P;
+use GuzzleHttp\Promise\PromiseInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Throwable;
+
+class RetryMiddleware
+{
+    private const RETRY_STATUS_CODES = [408, 429];
+
+    /**
+     * @var callable(RequestInterface, array): PromiseInterface
+     * @phpstan-ignore missingType.iterableValue
+     */
+    private $nextHandler;
+
+    /**
+     * @var array{
+     *     maxRetries: int,
+     *     baseDelay: int,
+     * }
+     */
+    private array $options;
+
+    /**
+     * @param callable $nextHandler
+     * @param array{
+     *     maxRetries?: int,
+     *  }|null $options
+     */
+    public function __construct(
+        callable $nextHandler,
+        ?array   $options = null,
+    ) {
+        $this->nextHandler = $nextHandler;
+        $this->options = array_merge([
+            'maxRetries' => 2,
+            'baseDelay' => 1000,
+        ], $options ?? []);
+    }
+
+    /**
+     * @param array{
+     *     maxRetries?: int,
+     *     baseDelay?: int,
+     * }|null $options
+     * @return callable
+     */
+    public static function create(?array $options = null): callable
+    {
+        return static function (callable $handler) use ($options): RetryMiddleware {
+            return new RetryMiddleware($handler, $options);
+        };
+    }
+
+    /**
+     * @param RequestInterface $request
+     * @param array{
+     *      retries?: int,
```

Comment:
Can we update this to be `retryAttempt` so it's more clear it's the attempt #? Otherwise it's confusing how it relates to `maxRetries`.

Reply:
Done